lib/generators/metka/strategies/materialized_view/templates/migration.rb.erb

```diff
@@ -5,41 +5,48 @@ class <%= @migration_class_name %> < ActiveRecord::Migration<%= ActiveRecord::VE
     execute <<-SQL
     CREATE OR REPLACE FUNCTION metka_refresh_<%= view_name %>_materialized_view() RETURNS trigger LANGUAGE plpgsql AS $$
       BEGIN
-        IF TG_OP = 'INSERT' AND NEW.<%= source_column_name %> IS NOT NULL THEN
+        IF TG_OP = 'INSERT' AND
+           (<%= source_columns.map { |column| "NEW.#{column} IS NOT NULL" }.join(' OR ') %>) THEN
           REFRESH MATERIALIZED VIEW CONCURRENTLY <%= view_name %>;
         END IF;
 
-        IF TG_OP = 'UPDATE' AND OLD.<%= source_column_name %> != NEW.<%= source_column_name %> THEN
+        IF TG_OP = 'UPDATE' AND
+           (<%= source_columns.map { |column| "OLD.#{column} IS DISTINCT FROM NEW.#{column}" }.join(' OR ') %>) THEN
           REFRESH MATERIALIZED VIEW CONCURRENTLY <%= view_name %>;
         END IF;
 
-        IF TG_OP = 'DELETE' AND OLD.<%= source_column_name %> IS NOT NULL THEN
+        IF TG_OP = 'DELETE' AND
+           (<%= source_columns.map { |column| "OLD.#{column} IS NOT NULL" }.join(' OR ') %>) THEN
           REFRESH MATERIALIZED VIEW CONCURRENTLY <%= view_name %>;
         END IF;
         RETURN NEW;
       END $$;
 
     DROP MATERIALIZED VIEW IF EXISTS <%= view_name %>;
     CREATE MATERIALIZED VIEW <%= view_name %> AS
-      SELECT UNNEST
-        ( <%= source_column_name %> ) AS <%= source_column_name.singularize %>_name,
-        COUNT ( * ) AS taggings_count
-      FROM
-        <%= source_table_name %>
```

Comment:
That's really an interesting moment: While using UNNEST directly within a SELECT, I was getting that kind of Query Plan:

```
GroupAggregate  (cost=23497.64..27285.14 rows=152500 width=40) (actual time=8.448..10.740 rows=100 loops=1)
  Group Key: (unnest((constructs.tags || constructs.materials)))
  ->  Sort  (cost=23497.64..23997.64 rows=200000 width=32) (actual time=8.415..9.299 rows=11975 loops=1)
        Sort Key: (unnest((constructs.tags || constructs.materials)))
        Sort Method: quicksort  Memory: 946kB
        ->  ProjectSet  (cost=0.00..1100.00 rows=200000 width=32) (actual time=0.014..3.171 rows=11975 loops=1)
              ->  Seq Scan on constructs  (cost=0.00..80.00 rows=2000 width=117) (actual time=0.008..0.348 rows=2000 loops=1)
Planning Time: 0.170 ms
Execution Time: 10.816 ms
```
The most disturbing part (except for the 10.816ms) is this: `GroupAggregate  (cost=23497.64..27285.14 rows=152500 width=40)`, > rows=152500, while UNNEST should actually show only 11974 rows